packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -440,10 +452,12 @@ If you provided a React element for the optionText prop, you must also provide t
             }
 
             return eventType && eventType === 'change'
-                ? selectedItemTexts.includes(filter) && selectedChoice
+                ? selectedItemTexts.includes(filter) &&
+                      selectedChoice &&
+                      selectedChoice[optionValue || 'id'] !== emptyValue
```

Comment:
Not sure this will work in the `multiple` case, because int this case `selectedChoice` seems to be an array (see line 447)

Reply:
Doesn't seems to be a problem for multiple, so I'll just make that evaluation when is not multiple